### DISCUSSION TITLE
Error handling when using useAsyncData and lazy loading

### DISCUSSION THREAD

**Discussion by johnrich85** [nuxt/nuxt#23387] [Questions]
I'm currently on 3.5.3 and it seems that when using SSR, the following throws an error correctly when hard refreshing the page. However, when browsing to the page from another (i.e. when rendering on the client) it does not.

```
const {data, error, pending} = await useAsyncData(
    async () => await $fetch('https://fakestoreapi.com/123123123'),
    {
      lazy: true
    }
)

if (error.value) {
  throw createError({ statusCode: 500, statusMessage: 'abc', fatal:true })
}
```

Does anyone know why this happens?

**Comment by shunnNet** [comment]
I think the reason is that you use `lazy: true`.
Nuxt will wait the request complete only at server when `lazy: true`, so when request completed and is failed (at server), we got an error with non null value. So the `createError()` is thrown.

But, at client side, the `useAsyncData` promise will immediately resolved (because `lazy: true`), at the time, the request haven't completed. So `error.value` will be null in setup. 

That's why throw error part not be executed at client side.

**Reply by obulat** [reply]
How can you execute the `throw createError` part on the client side to redirect to the `error.vue` page? I tried using a watcher to watch `error` value, but then I get the Vue warning that an error was thrown inside a watcher.
